### DISCUSSION TITLE
NextJS Image Dynamic/Automatic size/height?

### DISCUSSION THREAD

**Discussion by Torbet** [vercel/next.js#24260] [Help]
I'm using react markdown to render markdown files, and am passing images into an image component that I made. I'm trying to switch it to Next Image, but I can't get dynamic sizing for the original image. I'm not sure if its even possible, but it should be. Layout fill doesn't constrain to the max width and even if I wrap it all in a div, the height is just off. Is there anyway to pass lots of different image sizes into it and have them rendered correctly?

**Comment by akellbl4** [comment]
Hi,
You can check out @leerob's site source code.
He is using `image-size` to get image size and then replacing a default image tag with `next/image` in markdown.
This examples should help with your problem: [here](https://github.com/leerob/leerob.io/blob/main/scripts/img-to-jsx.js), [here](https://github.com/leerob/leerob.io/blob/main/scripts/markdown-to-next-image.js) and [here](https://github.com/leerob/leerob.io/blob/main/lib/mdx.js)
